core/src/main/java/org/infinispan/util/concurrent/BlockingManagerImpl.java

```diff
@@ -47,10 +52,19 @@ protected void start() {
 
    @Override
    public CompletionStage<Void> runBlocking(Runnable runnable, Object traceId) {
-      return runBlockingOperation(runnable, traceId, blockingExecutor);
+      return runBlockingOperation(runnable, traceId, blockingExecutor, true);
+   }
+
+   @Override
+   public void runBlockingAsync(Runnable runnable, Object traceId) {
+      CompletionStage<Void> stage = runBlockingOperation(runnable, traceId, blockingExecutor, false);
+      if (stageConsumer != null) {
```

Comment:
why the `BiConsumer`? Wouldn't be simpler just to check `trace`?
```
if (trace) {
         stage.whenComplete((ignored, t) -> log.tracef(t, "There was an exception during processing of a command with id %s", traceId));
      }
```
Also, shouldn't be checking if the throwable is `null`?

Reply:
The BiConsumer avoids the allocation every time when tracing is enabled. Figured it was simple enough to do and applied to all cases.

I agree that yes it should be checking null for the throwable.

Reply:
Ah originally I was using this in a couple places, but after I refactored the code it is only 1 spot. So I will move the trace to be inline with the caller and take the allocation per call.

Reply:
well, the `BiConsumer` is allocating another `BiConsumer` inside :) 
so, it will be the same.

Reply:
Yes, sorry, the point was originally to store the inner one in the variable. Refactoring gone wrong :)